mobile/src/main/java/org/openhab/habdroid/ui/PreferencesActivity.kt

```diff
@@ -348,15 +350,46 @@ class PreferencesActivity : AbstractBaseActivity() {
                 ?: preferenceScreen.sharedPreferences.getInt(Constants.PREV_SERVER_FLAGS, 0)
 
             if (flags and ServerProperties.SERVER_FLAG_ICON_FORMAT_SUPPORT == 0) {
-                val iconFormatPreference = preferenceScreen.findPreference(Constants.PREFERENCE_ICON_FORMAT)
                 preferenceScreen.removePreferenceFromHierarchy(iconFormatPreference)
+            } else {
+                iconFormatPreference.setOnPreferenceChangeListener { pref, _ ->
+                    val context = pref.context
+                    clearImageCache(context)
+                    val intent = Intent(context, ItemUpdateWidget::class.java)
+                    intent.action = AppWidgetManager.ACTION_APPWIDGET_UPDATE
+                    val ids = AppWidgetManager.getInstance(context)
+                        .getAppWidgetIds(ComponentName(context, ItemUpdateWidget::class.java))
+                    intent.putExtra(AppWidgetManager.EXTRA_APPWIDGET_IDS, ids)
+                    context.sendBroadcast(intent)
```

Comment:
```
val widgetComponent = ComponentName(context, ItemUpdateWidget::class.java)
val ids = AppWidgetManager.getInstance(context).getAppWidgetIds(widgetComponent)
val intent = Intent(widgetComponent)
    .setAction(AppWidgetManager.ACTION_APPWIDGET_UPDATE)
    .putExtra(AppWidgetManager.EXTRA_APPWIDGET_IDS, ids)
context.setBroadcast(intent)
```

Reply:
`ComponentName` cannot be passed to `Intent()`.

Reply:
I've optimized the `setAction` and `putExtra` calls.

Reply:
Looks good now. I was totally sure `Intent` has a constructor taking a `ComponentName`, because the `Intent(Context, Class<?>)` one also creates a ComponentName internally, but obviously I was wrong.